### PR TITLE
Add support for %load magic

### LINKED ISSUE #581: %load doesn't work in notebook or console
Typing `%load <file-or-url>` in the notebook or console should replace the current input with the contents and leave the `load` line commented out, e.g. this:

image

becomes:

image

That uses a special api call to reset the current input, it probably isn't implemented yet.

### PR DESCRIPTION
**Pull request by blink1073**
Fixes #581.

### CODE REVIEW

File: src/notebook/notebook/actions.ts [hunk 1]
```diff
@@ -880,6 +883,39 @@ namespace Private {
...
+      let text = (setNextInput as any).text;
+      let replace = (setNextInput as any).replace;
+      if (replace) {
+        child.model.source = text;
+      } else {
+        // Create a new code cell and add as the next cell.
+        let cell = parent.model.factory.createCodeCell();
+        cell.source = text;
+        let cells = parent.model.cells;
+        for (let i = 0; i < cells.length; i++) {
```
Comment (jasongrout) [inline src/notebook/notebook/actions.ts]: `parent.model.cells.indexOf`?

Comment (blink1073) [inline src/notebook/notebook/actions.ts]: Ha, yep.

File: src/notebook/notebook/actions.ts [hunk 2]
```diff
@@ -880,6 +883,34 @@ namespace Private {
...
+      let text = (setNextInput as any).text;
+      let replace = (setNextInput as any).replace;
+      if (replace) {
+        child.model.source = text;
+      } else {
+        // Create a new code cell and add as the next cell.
+        let cell = parent.model.factory.createCodeCell();
+        cell.source = text;
+        let cells = parent.model.cells;
+        let i = cells.indexOf(child.model);
```
Comment (jasongrout) [inline src/notebook/notebook/actions.ts]: There's a `-1` failure case to manage too (not likely, but still, could happen).

Comment (blink1073) [inline src/notebook/notebook/actions.ts]: This *is* the cell we are currently executing, but I admire your defensive stance.

Comment (jasongrout) [inline src/notebook/notebook/actions.ts]: This is a utility function that could be called by whomever, whenever :). Our convention is to call it from the running cell....

**Comment by jasongrout** [comment]
Looks great to me, and works well. Thanks!

**Comment by blink1073** [comment]
Nice, 75% complete on the 0.9 Milestone, w00t!

**Pull request merged**